delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,292 @@
+
+  /**
+    * Fetch the last version of a resolver
+    * @param id      the resolver identifier
+    * @param project the project where the resolver belongs
+    */
+  def fetch(id: Iri, project: ProjectRef): UIO[Option[ResolverResource]]
+
+  /**
+    * Fetches the resolver at a given revision
+    * @param id      the resolver identifier
+    * @param project the project where the resolver belongs
+    * @param rev     the current revision of the resolver
+    */
+  def fetchAt(id: Iri, project: ProjectRef, rev: Long): IO[RevisionNotFound, Option[ResolverResource]]
+
+  /**
+    * Fetches a resolver by tag.
+    *
+    * @param id        the resolver identifier
+    * @param project   the project where the resolver belongs
+    * @param tag       the tag revision
+    */
+  def fetchBy(
+      id: Iri,
+      project: ProjectRef,
+      tag: Label
+  ): IO[ResolverRejection, Option[ResolverResource]]
+
+  /**
+    * A non terminating stream of events for resolvers. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def events(offset: Offset = NoOffset): Stream[Task, Envelope[ResolverEvent]]
+
+}
+
+object Resolvers {
+
+  /**
+    * The resolvers module type.
+    */
+  final val moduleType: String = "resolver"
+
+  import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils.instant
+
+  private[delta] def next(state: ResolverState, event: ResolverEvent): ResolverState = {
+    (state, event) match {
+      // Creating a resolver
+      case (Initial, e: ResolverCreated)                                  =>
+        Current(
+          id = e.id,
+          project = e.project,
+          value = e.value,
+          tags = Map.empty,
+          rev = e.rev,
+          deprecated = false,
+          createdAt = e.instant,
+          createdBy = e.subject,
+          updatedAt = e.instant,
+          updatedBy = e.subject
+        )
+      // Updating a resolver
+      case (c: Current, e: ResolverUpdated) if c.value.tpe == e.value.tpe =>
+        c.copy(
+          value = e.value,
+          rev = e.rev,
+          updatedAt = e.instant,
+          updatedBy = e.subject
+        )
+
+      // Tagging a resolver
+      case (c: Current, e: ResolverTagAdded)                              =>
+        c.copy(rev = e.rev, tags = c.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+
+      // Deprecating a resolver
+      case (c: Current, e: ResolverDeprecated)                            =>
+        c.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+
+      // Cases that should happen and that should have been filtered out by the evaluation
+      case _                                                              => state
+    }
+  }
+
+  private[delta] def evaluate(state: ResolverState, command: ResolverCommand)(implicit
```

Comment:
I got used to the way we have done it for all the other state machines with inner methods...it helps with the readability (and then you don't need many of those comments). Many times when you need a comment, you can better do with a method :)

```scala
def evaluate(state: ResolverState, command: ResolverCommand) = {
   def create(c: CreateResolver): IO[ResolverRejection, ResolverCreated] = ???
   def update(c: UpdateResolver): IO[ResolverRejection, ResolverUpdated] = ???
   def tag(c: TagResolver): IO[ResolverRejection, ResolverTagAdded] = ???
   def(c: DeprecateResolver): IO[ResolverRejection, ResolverDeprecated] = ???

   command match {
      case c: CreateResolver    => create(c)
      case c: UpdateResolver    => update(c)
      case c: TagResolver       => tag(c)
      case c: DeprecateResolver => deprecate(c)
    }

}
```